users/tests/test_pipeline.py

```diff
@@ -0,0 +1,30 @@
+import mock
+from model_mommy import mommy
+from social_django.utils import load_backend, load_strategy
+
+from battles.models import Battle
+from common.utils.tests import TestCaseUtils
+from users.auth_pipeline import _handle_invite
+
+
+class TestInviteSignup(TestCaseUtils):
+
+    def setUp(self):
+        super().setUp()
+        self.strategy = load_strategy()
+        self.backend = load_backend(strategy=self.strategy, name='google-oauth2', redirect_uri='/')
+
+    @mock.patch('users.auth_pipeline.validate_invite_key')
+    def test_user_signed_up_with_invite(self, validate_key):
+        mommy.make('battles.Invite', key=123)
+        validate_key.return_value = setattr(self.user, 'has_invite', True)
```

Comment:
I was afraid of that. I don't know how to mock a method from the function (instead of an attribute), will take a look into the documentation this afternoon.